smdebug/tensorflow/keras.py

```diff
@@ -113,7 +115,13 @@ def _get_matching_collections(
                 continue
 
             if match_inc(ts_name, current_coll.include_regex):
-                if not current_coll.has_tensor(tensor):
+                # In TF 2.x, we can't put tensors in a set/dictionary as tensor.__hash__()
+                # is no longer available. tensor.experimental_ref() returns a hashable reference
+                # object to this Tensor.
+                check_tensor = tensor
+                if version.parse(tf.__version__) >= version.parse("2.0.0"):
+                    check_tensor = tensor.experimental_ref()
```

Comment:
Any gotchas with this method?

Reply:
there's a warning that this can be changed or removed. https://www.tensorflow.org/api_docs/python/tf/Tensor#experimental_ref. but there's no other way to hash a tensor